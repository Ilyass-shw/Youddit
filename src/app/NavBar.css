.sideNav {
	width: 4.4rem;
	font-size: 0.6em;
	position: fixed;
	margin-top: 0.5rem;
	color: #909090;
}

.sideNav ul {
	padding-left: 0;
	margin: 0;
	list-style-type: none;

}

.sideNav ul p {
	height: 4.55rem;
	display: flex;
	justify-content: center;
	align-items: center;
	flex-direction: column;
	transition: 0.018s;
}
.sideNav ul p:hover:not(a:first-child) {
	background-color: #404040;
}
.sideNav ul p:first-child {
	height: 3.6rem;
}
.sideNav ul li:first-child svg:hover {
	padding: 6px;
	background-color: #404040;
	border-radius: 50%;
}
