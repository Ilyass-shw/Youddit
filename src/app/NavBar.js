import React from "react";
import "./NavBar.css";
import { Icon } from "@iconify/react";
import bxMenu from "@iconify-icons/bx/bx-menu";
import homeIcon from "@iconify-icons/entypo/home";
import sharpLocalFireDepartment from "@iconify-icons/ic/sharp-local-fire-department";
import baselineSubscriptions from "@iconify-icons/ic/baseline-subscriptions";
import baselineVideoLibrary from "@iconify-icons/ic/baseline-video-library";


const NavBar = (props) => {
	return (
		<div className="sideNav">
			<ul>
				<a href="">
					<li>
						<Icon icon={bxMenu} color={"white"} width="1.5rem" height="1.5rem" />
					</li>
				</a>

				<a href="">
					<Icon icon={homeIcon} color={"#909090"} width="1.2rem" height="1.2rem" />
					<li>Home</li>
				</a>

				<a href="">
					<Icon icon={sharpLocalFireDepartment} color={"#909090"} width="1.2rem" height="1.2rem" />
					<li>Trending</li>
				</a>

				<a href="">
					<Icon icon={baselineSubscriptions} color={"#909090"} width="1.2rem" height="1.2rem" className="menu-icon" />
					<li>Subscriptions</li>
				</a>

				<a href="">
					<Icon icon={baselineVideoLibrary} color={"#909090"} width="1.5rem" height="1.5rem" className="menu-icon" />
					<li>Library</li>
				</a>
			</ul>
		</div>
	);
};

export default NavBar;
