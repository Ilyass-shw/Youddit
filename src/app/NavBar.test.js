import React from "react";
import NavBar from "./NavBar";
import { jest } from "@jest/globals";
import { render } from "@testing-library/react";


jest.mock("@iconify-icons/bx/bx-menu", () => {
	return {
		default: () => {
			return "rock";
		},
	};
});

jest.mock("@iconify/react", () => {
	return { Icon: () => <p>Icon mock</p> };
});

jest.mock("@iconify-icons/entypo/home", () => {
	return {
		default: () => {
			return "rock";
		},
	};
});

jest.mock("@iconify-icons/ic/sharp-local-fire-department", () => {
	return {
		default: () => {
			return "rock";
		},
	};
});
jest.mock("@iconify-icons/ic/baseline-subscriptions", () => {
	return {
		default: () => {
			return "rock";
		},
	};
});
jest.mock("@iconify-icons/ic/baseline-video-library", () => {
	return {
		default: () => {
			return "rock";
		},
	};
});

describe("NavBar", () => {
	it("should match the snapshot", () => {
		const { container } = render(<NavBar />);
		expect(container).toMatchSnapshot();
	});

	it("should render without crashing", () => {
		const div = document.createElement("div");
		render(<NavBar />, div);
	});
});
