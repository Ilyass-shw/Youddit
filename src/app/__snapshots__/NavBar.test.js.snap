// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
<div>
  <div
    class="sideNav"
  >
    <ul>
      <a
        href=""
      >
        <li>
          <p>
            Icon mock
          </p>
        </li>
      </a>
      <a
        href=""
      >
        <p>
          Icon mock
        </p>
        <li>
          Home
        </li>
      </a>
      <a
        href=""
      >
        <p>
          Icon mock
        </p>
        <li>
          Trending
        </li>
      </a>
      <a
        href=""
      >
        <p>
          Icon mock
        </p>
        <li>
          Subscriptions
        </li>
      </a>
      <a
        href=""
      >
        <p>
          Icon mock
        </p>
        <li>
          Library
        </li>
      </a>
    </ul>
  </div>
</div>
`;
